test/utils/InstanceTest.sol

```diff
@@ -458,4 +496,128 @@ abstract contract InstanceTest is HyperdriveTest {
             config.shareTolerance
         );
     }
+
+    /// @dev Fuzz Test to ensure deposit accounting is correct for opening longs
+    ///      with the share token.
+    /// @param basePaid Amount in terms of base to open a long.
+    function test_open_long_with_shares(uint256 basePaid) external {
+        // Get balance information before opening a long.
+        (
+            uint256 totalBaseSupplyBefore,
+            uint256 totalSharesSupplyBefore
+        ) = getSupply();
```

Comment:
This looks like the wrong order (above `shares` come before `base`). FWIW, this order is more intuitive to me (`base` and then `shares`), but regardless it looks like this needs to be updated.

Reply:
Why did this and the below test pass with this order?

Reply:
Agreed on base then shares, I confused my own self haha. This is actually how I implemented the override functions so the test cases are passing.